### PR TITLE
C/Service: first prototypes of service API Phase 1

### PR DESCRIPTION
**Pull request by myungjoo**
This is the start of designing https://github.com/nnstreamer/nnstreamer/issues/3473
Phase 1 APIs are declared.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
PTAL: @jaeyun-jung @again4you @gichan-jang @anyj0527 This is the first draft of ML-Service-API prototype. Please review this carefully.

File: c/include/ml-api-common.h [hunk 1]
```diff
@@ -98,6 +140,20 @@ typedef enum _ml_tensor_type_e
...
+ *          So developers should release the pipeline handle and recreate it again.
+ * @since_tizen 6.0
+ * @remarks The @a in can be used only in the callback. To use outside, make a copy.
+ * @remarks The @a out can be used only in the callback. To use outside, make a copy.
+ * @param[in] in The handle of the tensor input (a single frame. tensor/tensors).
+ * @param[out] out The handle of the tensor output to be filled (a single frame. tensor/tensors).
+ * @param[in,out] user_data User application's private data.
+ * @return @c 0 on success. @c 1 to ignore the input data. Otherwise a negative error value.
+ */
+typedef int (*ml_custom_easy_invoke_cb) (const ml_tensors_data_h in, ml_tensors_data_h out, void *user_data);
```
Comment (jaeyun-jung) [inline c/include/ml-api-common.h:155]: custom-easy is now used only in pipeline, so I think this is not a common callback. declaration.

Comment (myungjoo) [reply c/include/ml-api-common.h:155]: ML-Service-API may register a custom-easy function.

File: c/include/ml-api-service.h [hunk 2]
```diff
@@ -28,6 +33,88 @@
...
+  char *provider_name;
+  char *pipeline_description /**< This pipeline description should be compatible with ml_pipeline_construct API */
+  
+} ml_service_pipeline_description;
+
+/**
+ * @brief TBU
+ * @param[in] desc provider_type of desc is restricted to ML_SERV_PROV_APP or ML_SERV_PROV_APP_SERVICE. If it is ML_SERV_PROV_UNSPECIFIED, it is considered to be ML_SERV_PROV_APP.
+ */
+int ml_service_pipeline_description_add (const ml_service_pipeline_description * desc);
```
Comment (jaeyun-jung) [inline c/include/ml-api-service.h:66]: Discussion: function names and params
```
int ml_service_add_pipeline (const char *name, const char *description);
int ml_service_get_pipeline (const char *name, char **description);
int ml_service_add_model (...);
int ml_service_add_model_series (...);
```

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**